external/llvm-project/mlir/include/mlir/Dialect/Async/IR/AsyncOps.td

```diff
@@ -27,6 +28,66 @@ include "mlir/Interfaces/SideEffectInterfaces.td"
 class Async_Op<string mnemonic, list<Trait> traits = []> :
     Op<AsyncDialect, mnemonic, traits>;
 
+def Async_LaunchOp :
+  Async_Op<"launch", [DeclareOpInterfaceMethods<CallOpInterface>,
+                      AttrSizedOperandSegments]> {
+  let summary = "Asynchronous launch operation";
+  let description = [{
+    The `async.launch` operation encapsulates a call to a 'kernel' func that
+    may run asynchronously from the caller. A 'async.token' is returned to
+    maintain ordered dependent execution.
+
+    The actual concurrency semantics depends on the dialect lowering to the
+    executable format. Fully sequential execution ("compute0" completes before
+    "compute1" starts) is a completely legal execution.
+
+    Because concurrent execution is not guaranteed, it is illegal to create an
+    implicit dependency from "compute1" to "compute0" (e.g. via shared global
+    state). All dependencies must be made explicit with async launch arguments
+    (i.e. `async.token`).
+
+   `async.launch` operation takes `async.token` dependencies and operands
+    separately, and starts execution of the kernel function only when all
```

Comment:
what does "separately" mean?  I think what you mean is that the launch operation may complete without the code in the associated function being executed.

Reply:
"Separately" here refers to two buckets: dependencies which are tokens, and operands which are values.

Reply:
The sentence is kind-of talking about 2 different things. 

But in async.execute all operands are wrapped in an async.value and the spec stipulates that the launch waits for dependency tokens AND operands. I suppose this could facilitate memcpy dependencies..

Reply:
I think you need to say this explicitly... "In contrast with async.execute, this op...